test/functions/accept-test.js

```diff
@@ -1,7 +1,321 @@
+      const id = "devuser";
+      const { key, algorithm } = DEV_CREDENTIALS[id];
+      const body = Object.assign({}, DEFAULT_POST_BODY);
+      delete body.image;
+
+      const result = await acceptPost({
+        httpMethod: "POST",
+        proto: "https",
+        host: "example.com",
+        port: 443,
+        path: "/prod/accept",
+        id,
+        key,
+        algorithm,
+        body
+      });
+
+      expect(result.statusCode).to.equal(400);
+      expect(JSON.parse(result.body).error).to.equal(
+        'Required "image" is missing'
+      );
+    });
+
+    it("accepts a properly authorized image submission", async () => {
+      const id = "devuser";
+      const { key, algorithm } = DEV_CREDENTIALS[id];
+      const imageContent = "1234";
+      const imageContentType = "image/jpeg";
+      const body = Object.assign({}, DEFAULT_POST_BODY, {
+        image: {
+          filename: "image.jpg",
+          contentType: imageContentType,
+          content: imageContent
+        }
+      });
+
+      const result = await acceptPost({
+        httpMethod: "POST",
+        proto: "https",
+        host: "example.com",
+        port: 443,
+        path: "/prod/accept",
+        id,
+        key,
+        algorithm,
+        body
+      });
+
+      expect(mocks.putObject.args[0][0]).to.deep.equal({
+        Bucket: CONTENT_BUCKET,
+        Key: `image-${requestId}`,
+        Body: new Buffer(imageContent),
+        // ContentEncoding: "7bit",
+        ContentType: imageContentType
+      });
+      expect(mocks.getQueueUrl.args[0][0]).to.deep.equal({
+        QueueName: QUEUE_NAME
+      });
+
+      const message = mocks.sendMessage.args[0][0];
+      const messageBody = JSON.parse(message.MessageBody);
+
+      expect(messageBody.id).to.equal(requestId);
+      expect(message.QueueUrl).to.equal(QueueUrl);
+      expect(result.statusCode).to.equal(201);
+    });
   });
 });
+
+const DEFAULT_POST_BODY = {
+  negative_uri: "https://example.com/negative",
+  positive_uri: "https://example.com/positive",
+  positive_email: "positive@example.com",
+  notes: "foobar",
+  image: {
+    filename: "image.jpg",
+    contentType: "image/jpeg",
+    content: "1234123412341234"
+  }
+};
+
+async function acceptPost({
+  httpMethod,
+  proto,
+  host,
+  port,
+  path,
+  id,
+  key,
+  algorithm,
+  body = DEFAULT_POST_BODY
+}) {
+  const { contentType, encodedBody } = buildBody(body);
+  const hawkResult = Hawk.client.header(
+    `${proto}://${host}:${port}${path}`,
+    httpMethod,
+    { credentials: { id, key, algorithm } }
+  );
+  const headers = {
+    Host: host,
+    "X-Forwarded-Port": port,
+    "Content-Type": contentType,
+    Authorization: hawkResult.header
+  };
+  return accept.post(
+    {
+      path,
+      httpMethod,
+      headers,
+      body: encodedBody,
+      requestContext: { path, requestId }
+    },
+    {}
+  );
+}
+
+function buildBody(data) {
```

Comment:
This is ugly and I probably should have mocked this out or used a library, but it helped me understand why binary submissions weren't working in AWS